contracts/interfaces/draft-IERC7786.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @dev Interface for ERC-7786 source gateways.
+ *
+ * See ERC-7786 for more details
+ */
+interface IERC7786GatewaySource {
+    /**
+     * @dev Event emitted when a message is created. If `outboxId` is zero, no further processing is necessary, and
+     * no {MessageSent} event SHOULD be expected. If `outboxId` is not zero, then further (gateway specific, and non
+     * standardized) action is required.
+     */
+    event MessageCreated(
+        bytes32 outboxId,
+        string sender, // CAIP-10 account ID
+        string receiver, // CAIP-10 account ID
+        bytes payload,
+        bytes[] attributes
+    );
+
+    /**
+     * @dev This event is emitted when a message, for which the {MessageCreated} event contains an non zero `outboxId`,
+     * received the required post processing actions, and was thus sent to the destination chain.
+     */
+    event MessageSent(bytes32 indexed outboxId);
+
+    /// @dev This error is thrown when a message creation fails because of an unsupported attribute being specified.
+    error UnsupportedAttribute(bytes4 selector);
+
+    /// @dev Getter to check whether an attribute is supported or not.
+    function supportsAttribute(bytes4 selector) external view returns (bool);
+
+    /**
+     * @dev Endpoint for creating a new message. If the message requires further (gateway specific) processing before
+     * it can be sent to the destination chain, then a non-zero `outboxId` must be returned. Otherwise, this the
+     * message MUST be sent and this function must return 0.
+     *
+     * * MUST emit a {MessageCreated} event.
+     * * SHOULD NOT emit a {MessageSent} event.
```

Comment:
Why should it not emit a `MessageSent` event if above it says that when returning zero the message must be sent.

Reply:
That is part of the ERC specification. 

- If the message requires post processing, a unique id is give to that message, otherwize 0 is used
- The MessageSent is used when post processing is done, in that case we say which message was processed using that (non 0) id.
- if no processing is needed, the message is sent automatically when its created. emitting `MessageSent(0)` would be useless (that is implicit by 0 being in MessageCreated) but also confusing (id 0 is not unique)

Reply:
Makes sense. Thanks for the clarification.